src/globus_cli/commands/collection/create/guest.py

```diff
@@ -0,0 +1,211 @@
+    login_manager: LoginManager,
+    *,
+    mapped_collection_id: uuid.UUID,
+    collection_base_path: str,
+    user_credential_id: uuid.UUID | None,
+    local_username: str | None,
+    contact_info: str | None | ExplicitNullType,
+    contact_email: str | None | ExplicitNullType,
+    default_directory: str | None | ExplicitNullType,
+    department: str | None | ExplicitNullType,
+    description: str | None | ExplicitNullType,
+    display_name: str,
+    enable_https: bool | None,
+    force_encryption: bool | None,
+    identity_id: str | None,
+    info_link: str | None | ExplicitNullType,
+    keywords: list[str] | None,
+    public: bool,
+    organization: str | None | ExplicitNullType,
+    user_message: str | None | ExplicitNullType,
+    user_message_link: str | None | ExplicitNullType,
+    verify: dict[str, bool],
+) -> None:
+    """
+    Create a GCSv5 Guest Collection
+    """
+    gcs_client = login_manager.get_gcs_client(
+        collection_id=mapped_collection_id,
+        include_data_access=True,
+        assert_entity_type=(EntityType.GCSV5_MAPPED,),
+    )
+
+    if not user_credential_id:
+        user_credential_id = _select_user_credential_id(
+            gcs_client, mapped_collection_id, local_username, identity_id
+        )
+
+    converted_kwargs: dict[str, t.Any] = ExplicitNullType.nullify_dict(
+        {
+            "collection_base_path": collection_base_path,
+            "contact_info": contact_info,
+            "contact_email": contact_email,
+            "default_directory": default_directory,
+            "department": department,
+            "description": description,
+            "display_name": display_name,
+            "enable_https": enable_https,
+            "force_encryption": force_encryption,
+            "identity_id": identity_id,
+            "info_link": info_link,
+            "keywords": keywords,
+            "mapped_collection_id": mapped_collection_id,
+            "public": public,
+            "organization": organization,
+            "user_credential_id": user_credential_id,
+            "user_message": user_message,
+            "user_message_link": user_message_link,
+        }
+    )
+    converted_kwargs.update(verify)
+
+    try:
+        res = gcs_client.create_collection(
+            globus_sdk.GuestCollectionDocument(**converted_kwargs)
+        )
+    except globus_sdk.GCSAPIError as e:
+        # Detect session timeouts related to HA collections.
+        # This is a hacky workaround until we have better GARE support across the CLI.
+        if _is_session_timeout_error(e):
+            endpoint_id = gcs_client.source_epish.get_collection_endpoint_id()
+            context = LoginContext(
+                error_message="Session timeout detected; Re-authentication required.",
+                login_command=f"globus login --gcs {endpoint_id} --force",
+            )
+            raise MissingLoginError([endpoint_id], context=context)
```

Comment:
Largely to make sure I'm following correctly:

This will force a new login against the endpoint (not the mapped collection). This will rectify session-related errors? As far as I can tell, there aren't new codepaths which will result in this passing session parameters. Are we just assuming that it's an authentication timeout of some kind?

Judging by the test case data, there's actually some possibility that this is a case in which a session parameter _would_ be appropriate, but our toolchain isn't sufficiently parsing it today. Is that about right?

Reply:
That's roughly right, yeah.

This relies on the fact that an HA collection cannot have a data_access scope so endpoint:manage_collections is all you could possibly need.

As far as detection criteria & wording, I posted this conditional & resulting text in the GCS channel to determine whether the "identities" check taking place is a sane approach to result in this wording & am waiting to hear back form them.

Reply:
I'm okay with this as an interim solution, even if we don't hear back (I believe they are short on personnel this week).

I would like us to get GARE parsing souped up enough to be able to handle these responses anyway, as your comment suggests, and then we can apply the relevant logic.

Reply:
Yeah, I took a look at potentially handling via expanded GARE parsing on Friday and I don't think it will be difficult to add in variants for the GCS errors I found, but it wasn't so trivial that I could knock it out (with tests) in a half hour right at EOD either.